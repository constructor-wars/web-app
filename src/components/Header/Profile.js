import React from 'react';
import { Link } from 'react-router-dom';

export default function Profile() {
  return (
    <div className="profile">
      <Link to="/profile">Profile</Link>
    </div>
  );
}
