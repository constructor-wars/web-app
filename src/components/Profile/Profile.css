.profileImg {
  height: 200px;
  width: 200px;
}

.profile_container {
  margin: 30px 300px;
  padding: 20px;
  border: 1px solid black;
  border-radius: 5px;
}
