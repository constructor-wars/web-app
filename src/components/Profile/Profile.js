import React from "react";
import "./Profile.css";
class Profile extends React.Component {
  render() {
    return (
      <div>
        <div className="profile__container">
          <img className="profileImg" src={this.props.profilepic} />
          <p>Username:{this.props.username}</p>
          <p>displayName:{this.props.displayName}</p>
          <a href={this.props.profileUrl}>Github url:{this.props.profileUrl}</a>
        </div>
      </div>
    );
  }
}

export default Profile;
