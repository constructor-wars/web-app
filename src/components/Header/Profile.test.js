import React from "react";
import Profile from "./Profile";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Profile", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("tests profile component", () => {
    const wrapper = shallow(<Profile />);
    const text = wrapper.find("div").text();
    expect(text).toEqual("Profile");
  });
});
