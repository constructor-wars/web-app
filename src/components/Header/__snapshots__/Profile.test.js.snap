// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches the snapshot 1`] = `
<div
  className="profile"
>
  Profile
</div>
`;
